Packs/PAN-OS/Playbooks/PAN-OS_-_Extract_IPs_From_Traffic_Logs_To_Sinkhole.yml

```diff
@@ -0,0 +1,922 @@
+      scriptName: PrintErrorEntry
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "73"
+    scriptarguments:
+      message:
+        simple: |-
+          Please make sure you run the "PAN-OS - Configure DNS Sinkhole" playbook, or configure a DNS sinkhole in PAN-OS manually, before executing this playbook.
+          This playbook failed either because an address object for the sinkhole address was not found (you should create an address object with FQDN "sinkhole.paloaltonetworks.com"), or there is no security rule that denies traffic to that address object.
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": -1090,
+          "y": 2295
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+view: |-
+  {
+    "linkLabelsPosition": {
+      "77_73_#default#": 0.13,
+      "79_64_yes": 0.24,
+      "80_64_yes": 0.27,
+      "80_84_#default#": 0.15,
+      "82_84_#default#": 0.19
+    },
+    "paper": {
+      "dimensions": {
+        "height": 2715,
+        "width": 1930,
+        "x": -1090,
+        "y": -60
+      }
+    }
+  }
+inputs:
+- key: OutputLogsForDetectedIPs
+  value:
+    simple: "False"
+  required: true
+  description: |-
+    Whether to output full traffic logs for the infected IPs.
+
+    If set to True, this will output the logs containing the traffic from the compromised IPs to the sinkhole address. It includes data that can be used for remediation or for deeper investigation by examining what traffic the hosts are creating.
+
+    If set to False, data from the logs will not be outputted.
+
+    Note: The IP addresses of the systems creating traffic to the sinkhole address will be outputted regardless of playbook input configurations.
+  playbookInputQuery:
+- key: RuleNameToDenyToSinkhole
+  value: {}
+  required: false
+  description: |-
+    The name of the rule that will be created  to deny traffic to the sinkhole address. This is required so that traffic logs will reveal the endpoints attempting to connect to their C&C (which by the end of this playbook will be sinkholed).
+    If a rule with the specified name does not exist, one will be created. The new rule will be placed before rules that allow DNS traffic, as recommended by the PAN-OS best practices.
+  playbookInputQuery:
+- key: TimeToWaitForTraffic
+  value:
+    simple: "300"
+  required: true
+  description: |-
+    The time in seconds that the playbook will wait for hosts to generate traffic to the sinkhole address.
+    Waiting for long periods of time will increase your chances of finding additional infected systems. However, if enough time passed since your DNS sinkhole was first configure, it may not be needed to wait at all, as systems are already generating traffic to the sinkhole address.
```

Comment:
```suggestion
    Waiting for long periods of time will increase your chances of finding additional infected systems. However, if enough time passed since your DNS sinkhole was first configured, it may not be needed to wait at all, as systems are already generating traffic to the sinkhole address.
```